# SOC 210B Assignment 7
_By M. Nolan Gray_

In [22]:
import statsmodels.api as sm
import pandas as pd
from statsmodels.formula.api import logit

In [16]:
dtafile = './Data/nlsw88_occ4.dta'
df = pd.read_stata(dtafile)
df.sample(5)

idcode  age   race  married  never_married  grade          collgrad  \
43        86   44  white   single              1   17.0      college grad   
170      403   40  white   single              1   12.0  not college grad   
907     2140   41  white   single              0   12.0  not college grad   
421      999   40  white   single              0   10.0  not college grad   
1991    4578   41  white  married              0   16.0      college grad   

      south     smsa  c_city                industry occupation     union  \
43        0     SMSA       0   Entertainment/Rec Svc   Laborers     union   
170       0     SMSA       0   Public Administration      Sales  nonunion   
907       0  nonSMSA       0  Wholesale/Retail Trade      Sales     union   
421       0  nonSMSA       0       Personal Services  Transport  nonunion   
1991      1  nonSMSA       0   Professional Services      Other       NaN   

           wage  hours    ttl_exp     tenure             occ4  
43    13.172291   44.0  16.269230   1.250000     lower manual  
170    5.732688   40.0  10.500000  10.500000  lower nonmanual  
907    8.067631   48.0  16.076923   0.000000  lower nonmanual  
421    6.038648   35.0   6.839744   2.666667     upper manual  
1991   3.873905   24.0  10.250000   1.500000            other

In [30]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2246 entries, 0 to 2245
Data columns (total 18 columns):
 #   Column         Non-Null Count  Dtype   
---  ------         --------------  -----   
 0   idcode         2246 non-null   int16   
 1   age            2246 non-null   int8    
 2   race           2246 non-null   category
 3   married        2246 non-null   category
 4   never_married  2246 non-null   int8    
 5   grade          2244 non-null   float64 
 6   collgrad       2246 non-null   category
 7   south          2246 non-null   int8    
 8   smsa           2246 non-null   category
 9   c_city         2246 non-null   int8    
 10  industry       2232 non-null   category
 11  occupation     2237 non-null   category
 12  union          1878 non-null   category
 13  wage           2246 non-null   float32 
 14  hours          2242 non-null   float64 
 15  ttl_exp        2246 non-null   float32 
 16  tenure         2231 non-null   float32 
 17  occ4           2237 non-null   ca

## Question 1

In [18]:
formula = ('never_married ~ collgrad + race')
model = logit(formula = formula, data = df).fit()

Optimization terminated successfully.
         Current function value: 0.319399
         Iterations 7


In [19]:
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:          never_married   No. Observations:                 2246
Model:                          Logit   Df Residuals:                     2242
Method:                           MLE   Df Model:                            3
Date:                Fri, 04 Mar 2022   Pseudo R-squ.:                 0.04424
Time:                        08:01:22   Log-Likelihood:                -717.37
converged:                       True   LL-Null:                       -750.57
Covariance Type:            nonrobust   LLR p-value:                 2.506e-14
============================================================================================
                               coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------
Intercept                   -2.7171      0.110    -24.710      0.000      -2.933      -2.502
collgrad[T.college grad]     0.7287      0.151      4.815      0.000       0.432       1.025
race[T.black]                1.0477      0.145      7.243      0.000       0.764       1.331
race[T.other]                0.3804      0.625      0.608      0.543      -0.845       1.606
============================================================================================
"""

Our log-odds formula is this: -2.72 + 0.73(college) + 1.05(black). I have left out race:other as it is not statistically significant. The default is a white woman with no college degree.

A white woman with a college degree:
* Log-odds: -1.99 (-2.72 + 0.73)
* Odds: 0.14 (e^-1.99/(1+e^-1.99)) 
* Probability: 0.12 (0.14/(1+0.14))

A black woman with a college degree:
* Log-odds: -0.94 (-2.72 + 0.73 + 1.05)
* Odds: 0.39 (e^-0.94/(1+e^-0.94))
* Probability: 0.28 (0.39/(1+0.39))

## Question 2

In [53]:
from statsmodels.formula.api import mnlogit
import statsmodels.formula.api as smf